transaction-service/src/main/java/pl/adamboguszewski/transaction/service/application/TransactionService.java

```diff
@@ -0,0 +1,34 @@
+package pl.adamboguszewski.transaction.service.application;
+
+import org.springframework.stereotype.Service;
+import pl.adamboguszewski.transaction.service.api.transaction.CreateTransactionRequest;
+import pl.adamboguszewski.transaction.service.application.dto.TransactionDto;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+// [TODO] Fill the details of this class.
+@Service
+public class TransactionService {
+
+    public List<Transaction> getAll() {
+        return Collections.emptyList();
+    }
+
+    public Optional<Transaction> getById(Long id) {
+        return Optional.empty();
+    }
+
+    public Optional<Transaction> createTransaction(TransactionDto dto) {
+        return Optional.of(Transaction.fromDto(dto));
+    }
+
+    public void replaceTransaction(Long id) {
+
```

Comment:
https://restfulapi.net/http-methods/#:~:text=If%20a%20new%20resource%20has,successful%20completion%20of%20the%20request.

> Use PUT APIs primarily to update existing resource (if the resource does not exist, then API may decide to create a new resource or not). If a new resource has been created by the PUT API, the origin server MUST inform the user agent via the HTTP response code 201 (Created) response and if an existing resource is modified, either the 200 (OK) or 204 (No Content) response codes SHOULD be sent to indicate successful completion of the request.

So first call getById, then depends of method return :
call createTransaction and return transaction
or do the update and return empty optional
then in controller return suitable http code (let's assume that everything goes right)